packages/datetime/src/dateRangeInput.tsx

```diff
@@ -211,17 +348,74 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
 
     private getSelectedRange = () => {
         return [this.state.selectedStart, this.state.selectedEnd].map((selectedBound?: moment.Moment) => {
-            return (!isMomentValidAndInRange(selectedBound, this.props.minDate, this.props.maxDate))
-                ? undefined
-                : fromMomentToDate(selectedBound);
+            return this.isMomentValidAndInRange(selectedBound)
+                ? fromMomentToDate(selectedBound)
+                : undefined;
         }) as DateRange;
     }
 
+    private getStartInputDisplayString = () => {
+        return this.state.isStartInputFocused
+            ? this.state.startInputString
+            : this.getFormattedDateString(this.state.selectedStart);
+    }
+
+    private getEndInputDisplayString = () => {
+        return this.state.isEndInputFocused
+            ? this.state.endInputString
+            : this.getFormattedDateString(this.state.selectedEnd);
+    }
+
     private getFormattedDateString = (momentDate: moment.Moment) => {
         if (isMomentNull(momentDate)) {
             return "";
         } else {
             return momentDate.format(this.props.format);
         }
     }
+
+    private dateStringToMoment = (dateString: String) => {
+        return moment(dateString, this.props.format);
+    }
+
+    private isMomentValidAndInRange = (momentDate: moment.Moment) => {
+        return isMomentValidAndInRange(momentDate, this.props.minDate, this.props.maxDate);
+    }
+
+    private getStateKeysAndValuesForBoundary = (boundary: DateRangeBoundary): IStateKeysAndValuesObject => {
+        const controlledValue = fromDateRangeToMomentDateRange(this.props.value);
+
+        if (boundary === DateRangeBoundary.START) {
+            return {
+                keys: {
+                    inputString: "startInputString",
+                    isInputFocused: "isStartInputFocused",
+                    selectedValue: "selectedStart",
+                },
+                values: {
+                    controlledValue: (controlledValue != null) ? controlledValue[0] : undefined,
+                    inputString: this.state.startInputString,
+                    isFocused: this.state.isStartInputFocused,
+                    selectedValue: this.state.selectedStart,
+                },
+            } as IStateKeysAndValuesObject;
+        } else if (boundary === DateRangeBoundary.END) {
+            return {
+                keys: {
+                    inputString: "endInputString",
+                    isInputFocused: "isEndInputFocused",
+                    selectedValue: "selectedEnd",
+                },
+                values: {
+                    controlledValue: (controlledValue != null) ? controlledValue[1] : undefined,
+                    inputString: this.state.endInputString,
+                    isFocused: this.state.isEndInputFocused,
+                    selectedValue: this.state.selectedEnd,
+                },
```

Comment:
The `DateRangeInputField` you mention in the description will be another step towards keeping consistencies between `DateInput` and `DateRangeInput` though I am wary we might get into another rabbit hole